lib/rules/yoda.js

```diff
@@ -245,11 +235,23 @@ module.exports = {
             function isBetweenTest() {
                 let leftLiteral, rightLiteral;
 
-                return (node.operator === "&&" &&
-                    (leftLiteral = getNormalizedLiteral(left.left)) &&
-                    (rightLiteral = getNormalizedLiteral(right.right, Number.POSITIVE_INFINITY)) &&
-                    leftLiteral.value <= rightLiteral.value &&
-                    same(left.right, right.left));
+                if (node.operator === "&&") {
+                    leftLiteral = getNormalizedLiteral(left.left);
+                    rightLiteral = getNormalizedLiteral(right.right);
+
+                    if (leftLiteral === null && rightLiteral === null) {
+                        return false;
+                    }
+
+                    if (leftLiteral !== null && rightLiteral === null || rightLiteral !== null && leftLiteral === null) {
+                        return true;
+                    }
```

Comment:
This could produce a false negative:

```js
/* eslint "yoda": ["error", "never", { "exceptRange": true }] */

if (0 < a && b < max) {} // this should be error
```

This function can return `true` only when `same(left.right, right.left)` is `true`, so we should probably move the `same` condition to the main `if (node.operator === "&&)`, as it was before.

(the same applies to `isOutsideTest`)

Reply:
done 👍